### PR TITLE
Testing Scala 2.13

### PR DESCRIPTION
**Pull request by aseure**
This PR includes all the changes from @dragisak on #553 plus some dependency updates.

### CODE REVIEW

**Comment by jlogeart** [comment]
@aseure Thank you for the change.

Are there any blockers that we can help with?

**Comment by aseure** [comment]
@jlogeart Thank you for your interest.

We've been delayed this release since last week because of our integration tests failing due to internal issues with our Algolia testing applications (used by our integration tests). We'll merge that as soon as we confirmed that our application is sane again and our tests are passing.

File: build.sbt [hunk 1]
```diff
@@ -33,22 +33,20 @@ libraryDependencies += "org.slf4j" % "slf4j-api" % slf4jVersion
...
-libraryDependencies += "ch.qos.logback" % "logback-classic" % logbackVersion % "test"
+libraryDependencies += "org.scalatest" %% "scalatest" % scalaTestVersion % Test
+libraryDependencies += "org.scalacheck" %% "scalacheck" % scalacheckVersion % Test
+libraryDependencies += "org.scalamock" %% "scalamock" % scalaMockVersion % Test
+libraryDependencies += "ch.qos.logback" % "logback-classic" % logbackVersion % Test
 
 scalacOptions ++= Seq(
   "-deprecation",
-  "-Xfatal-warnings",
+  // "-Xfatal-warnings",
```
Comment (Ant-hem) [inline build.sbt]: Why is this commented?

File: build.sbt [hunk 2]
```diff
@@ -6,21 +6,21 @@ description := "Scala client for Algolia Search API"
...
 coverageEnabled := false
 
-val asyncHttpClientVersion = "2.10.1"
-val json4sVersion = "3.5.2"
-val slf4jVersion = "1.7.25"
-val scalaUriVersion = "0.4.16"
+val asyncHttpClientVersion = "2.10.4"
+val json4sVersion = "3.6.7"
+val slf4jVersion = "1.7.30"
+val scalaUriVersion = "1.4.10"
```
Comment (Ant-hem) [inline build.sbt]: No BC upgrading to a new major version of this dependency?

File: build.sbt [hunk 3]
```diff
@@ -33,22 +33,20 @@ libraryDependencies += "org.slf4j" % "slf4j-api" % slf4jVersion
...
 
 scalacOptions ++= Seq(
   "-deprecation",
-  "-Xfatal-warnings",
+  // "-Xfatal-warnings",
   "-feature", //Emit warning and location for usages of features that should be imported explicitly.
   "-encoding",
   "UTF-8",
   "-unchecked", //Enable additional warnings where generated code depends on assumptions
-  "-Yno-adapted-args",
```
Comment (Ant-hem) [inline build.sbt]: Why did we delete this one?

Comment (aseure) [reply build.sbt]: Removed in [Scala 2.13](https://github.com/scala/scala/releases/tag/v2.13.0):

Screenshot 2020-01-24 at 11 56 19

File: build.sbt (hunk 2 above)
Comment (aseure) [reply build.sbt]: So far so good 🤞

File: build.sbt (hunk 1 above)
Comment (aseure) [reply build.sbt]: Indeed, we want to keep it. This took me more time than expected. See https://github.com/algolia/algoliasearch-client-scala/pull/571/commits/e86a20b3a888df227dd24bb54c3a9490b93487a1 cc @Ant-hem @dragisak

Comment (dragisak) [reply build.sbt]: You get warning because `scala.collection.JavaConverters` are deprecated in Scala 2.13

**Comment by aseure** [comment]
@Ant-hem If you could review/approve on last time the changes themselves. I've answered to your 3 comments. The CI is not passing because of flaky tests which do not happen when I run them locally. I'll try to fix them but I assume it has to do with poor network in TravisCI so all retries fail randomly on some methods.

**Pull request merged**